R/draft.R

```diff
@@ -132,7 +132,40 @@ draft <- function(file,
   invisible(file)
 }
 
+#' List available R Markdown template in a package
+#'
+#' List the available template by name that can be used with [draft()] to create
+#' a new document for R Markdown from a package.
+#'
+#' @param package Package to list template from. Default to **rmarkdown**
+#' @param full_path Set to `TRUE` to get the full path to the available templates
+#'
+#' @return A character vector of templates name to use within [draft()]. If
+#'   `full_path = TRUE`, it returns the full path to the templates.
+#' @export
+#' @md
+#' @examples
+#' # List rmarkdown templates & create a draft
+#' available_templates()
+#'
+#' # List rticles templates
+#' \dontrun{
+#'   available_templates("rticles")
+#' }
+available_templates <- function(package = "rmarkdown", full_path = FALSE) {
+  template_folder <- pkg_file("rmarkdown", "templates", package = package)
+  if (dir_exists(template_folder)) {
+    templates <- list.files(template_folder, full.names = TRUE)
+    # exclude file in folder if any
+    templates <- templates[utils::file_test('-d', templates)]
+    if (full_path) return(templates)
+    basename(templates)
+  }
```

Comment:
It was to return NULL. I should have written
````
if (!dir_exists(template_folder)) return(invisible(NULL))
````

Is returning `character(0)` instead better ?
``` r
available_templates("donotexist")
#> character(0)
```

Reply:
I don't care if it returns `NULL` or `character(0)` in this case. The latter is slightly better if type is important (i.e., the function returns character consistently), but in this case, I don't think it's important at all.

It seems you prefer `invisible()` value if the template folder is not found, which also makes sense.

I would slightly prefer saving one `if` statement, since it makes the code slightly simpler. Anyway, it's totally up to you.

Reply:
I prefer to check for this and not execute the remaining code if folder does not exists. I simplified by doing an early exist instead of wrapping whole code in if structure. And yes I would prefer invisible.

Thanks for the feedback